src/v2/controllers/FormController.js

```diff
@@ -147,7 +149,19 @@ export default Controller.extend({
     }
 
     this.toggleFormButtonState(true);
+
+    // NOTE: okta-idx-js does not know whether remediations
+    // are required to redirect or perform a GET request.
+    // As a result, we handle this accordingly by checking the formName
+    const fullPageRedirectForms = [ FORMS.REDIRECT_IDP ];
+    if (fullPageRedirectForms.includes(formName)) {
+      const currentViewState = this.options.appState.getCurrentViewState();
+      Util.redirectWithFormGet(currentViewState.href);
```

Comment:
why not only do this in the view that requires it ? instead of here ? for eg `SuccessView.js` does that

Reply:
Main reason is to centralize the redirect logic to the IDX handler, as opposed to having multiple implementations for each View/Controller. Keeping this logic at the `FormController` layer ensures any remediation response that contains `redirect-idp` will redirect when a user interacts with the **save** button.

Reply:
What are the other use cases that sends `redirect-idp` ?

Reply:
better to create a separate handler or function? @nikhilvenkatraman-okta mentioned it as TODO https://oktainc.atlassian.net/browse/OKTA-250473

Reply:
I'm pushing back on adding extra complexity for developers - as we're essentially forking here when processing the IDX response. Originally, I went down the path of adding a new handler, but quickly found that this requires:

- Additional model events need to be subscribed to and handled
- Disable the default `saveForm` function in favor of the new behavior

@haishengwu-okta - Would appreciate any input you have here!

Reply:
- For your case, maybe we can follow same pattern as other IDP buttons added to identify view? `FormInputFactory.createIdpButtons`?
- But, a generic redirect handler for ion form is still needed. There are a few cases as Pradeep point out (hence the JIRA). Don't have a good idea now. We can have a group discussion. What Jordan have done seems good.

Reply:
Another idea:
- Looks like for those redirect case, the remediation form doesn't have `rel: [ 'create-form' ]`
- Maybe confirm with IDX and Device team (has redirect flow for OktaVerify as well) to see if it is always like this.
- IF so, we can change idx-js
- Otherwise,  🤔

Reply:
Thanks for the feedback! From what I can see, we'll need to handle redirects in a couple of use cases currently:

1. Social login - where the user needs to interact with a custom IdP (Facebook, Google, etc.) as their primary authentication method.
2. When we're enrolling a factor (includes `select-authenticator-enroll` form) with a remediation of type `redirect-idp` (Custom claims, SSO Extension).
3. We have factor enrollment and verification where we encourage the user to enroll through a native application (similar to OV) using a universal link (opens up a native app on desktop/mobile devices). Normal `GET` requests unfortunately will not open up these applications, only full-page redirects.

While the `redirect-idp` type is closely related for both Social and custom factor enrollment, the login flows do not require an "enroll authenticator" page like the factors do. As a result, we need a generic solution to handle this type when we encounter it.

@swiftone and @haishengwu-okta - We _could_ make this change in `idx-js`, but I think we'd need a formal discussion around whether or not this is the best idea to have the library follow redirect requests. Specifically, this change will require us to modify our existing [fetch](https://github.com/okta/okta-idx-js/blob/master/src/v1/generateIdxAction.js#L19-L47) functionality to consider `redirect-idp` types.

@haishengwu-okta and @pradeepdewda-okta - If the current implementation is still not our ideal case, I'll set aside some time for us to chat today or tomorrow to figure out a different path forward. Sound good?

Reply:
Discussed February 9, 2021:

**tl;dr**: This implementation is fine for now. Will work with the `okta-idx-js` team to standardize on a solution for handling redirects in the near future.

Follow-up: Migrate `SuccessView` and and `SSOExtension` to use this centralized logic.